### ISSUE TITLE
Edit track points menu - Fast scroll

### ISSUE THREAD

**Issue by anastasiia936** [osmandapp/OsmAnd#13749]
1. Add options to "Delete all points before this point" and "Delete all points after this point" (as options available on a particular point).
 2. Please check the Scroll in the Points menu.  If there are a large number of dots, scrolling to the last dot slows down. In the video below, you can see that the scroll is slowing down.
 https://user-images.githubusercontent.com/80623554/153847655-3ef27b5a-6375-4995-9af7-4449522b2691.mp4
 3. Add an option to "Scroll to end of track", to jump there in one click

**Comment by AndyB69** [comment]
Hi, thanks Anastasia. 

Item 1 is a suggestion resulting from those occasions where you forget to stop the track at the correct place (at the end of a walk, say) and then jump in a car or train and it continues recording points!  I want a quick way to 'crop' all points _after_ a specific point. It can also happen at the start of a track but much more infrequently, I think, but may as well provide that option as well. 

It would also be helpful if by pointing/clicking at a point on the actual track line on the screen, that the list of points jumps to that point, rather than having to scroll the points list, which is extremely slow for even a medium length track, and as point 2 indicates has some processing efficiency issues. This would also resolve point 3.

Regards.

**Comment by Gbernardy** [comment]
> 1. Add options to "Delete all points before this point" and "Delete all points after this point" (as options available on a particular point).

This option is already present in the point panel

![IMG_20220220_214946](https://user-images.githubusercontent.com/99968793/154863801-fd9c5054-96a9-4bd0-9cad-86bbf05925cb.jpg)

**Comment by AndyB69** [comment]
Ah, somehow I'd not seen that, I see it now. Thank you.

What does "Split before" and "Split after" do - create two separate tracks
at those points?

Also - is there a way to position the scrolling list of points on the
correct point by clicking a point on the track itself (pointing at the
track line) OR by choosing a point in the Graph view? In Graph view I can
position a dot on the track, but I can't see how to locate the Point for
that dot - when I switch to Points view, irrespective of where I positioned
the dot in Graph view, it starts by showing me the first points in the
list.

Regards,
Andrew

**Comment by Gbernardy** [comment]
What does "Split before" and "Split after" do - create two separate tracks at those points?

Never tried ... I think so.


 Also - is there a way to position the scrolling list of points on the correct point by clicking a point on the track itself (pointing at the track line) OR by choosing a point in the Graph view? In Graph view I can position a dot on the track, but I can't see how to locate the Point for that dot - when I switch to Points view, irrespective of where I positioned the dot in Graph view, it starts by showing me the first points in the list. 

to open the point panel you have to:

Open the track
Edit the track
Click Short tap on point    click sul punto che tu vuoi visualizzare   point panel

**Comment by Gbernardy** [comment]
> Ah, somehow I'd not seen that, I see it now. Thank you. What does "Split before" and "Split after" do - create two separate tracks at those points? 

Never tried ... I think so.

Also - is there a way to position the scrolling list of points on the correct point by clicking a point on the track itself (pointing at the track line) OR by choosing a point in the Graph view? In Graph view I can position a dot on the track, but I can't see how to locate the Point for that dot - when I switch to Points view, irrespective of where I positioned the dot in Graph view, it starts by showing me the first points in the list.

No. When you have opened the list of points, if you click on a point of the track path (on graph), if short tap you open the point panel, if long tap you can move the point on the path of the track (on graph).

In my opinion, the points list should only be opened from the points panel, with the first point in the list being clicked on the track path.  With this solution, the scrool of the points list would also be realized.

**Comment by AndyB69** [comment]
Right, thanks. So apart from the remaining suggestions, I think this one
needs to be made as well:

   - In Graph view, when scrolling sideways in the chart, a visible point
   moves along the track itself. If one was to then switch to Point view it
   would be ideal if the Points list positioned itself *on the point
   selected in Graph view*.
      - At the moment, irrespective of whatever position you select in
      Graph, the Points list is always presented where you last
manually scrolled
      to in that list.
      - Right now, if you manually scroll to a point in the Points list,
      and click that point, this point is breifly indicated on the track itself
      with a dot, but that dot is immediately obscured by the pop-up
      point-editing box. And when you close that box, the point
indicator on the
      track itself disappears.
      - Whereas, the other dot that is shown on the track from selecting a
      position in Graph view does remain visible all the time, but
can't be used
      to position the Points list.
      - i.e. there is not a good connection between Graph track position
      and Points position. If they could be connected, that would be a massive
      improvement in usability for track editing. i.e. select a point via graph
      view and the Points list scrolls to that point, and in reverse, if you
      select a point on the Points list it could position the Graph
spot in that
      same position.

Regards,
Andrew

**Comment by Gbernardy** [comment]
> Right, thanks. So apart from the remaining suggestions, I think this one needs to be made as well: - In Graph view, when scrolling sideways in the chart, a visible point moves along the track itself. If one was to then switch to Point view it would be ideal if the Points list positioned itself *on the point selected in Graph view*. - At the moment, irrespective of whatever position you select in Graph, the Points list is always presented where you last manually scrolled to in that list. - Right now, if you manually scroll to a point in the Points list, and click that point, this point is breifly indicated on the track itself with a dot, but that dot is immediately obscured by the pop-up point-editing box. And when you close that box, the point indicator on the track itself disappears. - Whereas, the other dot that is shown on the track from selecting a position in Graph view does remain visible all the time, but can't be used to position the Points list. - i.e. there is not a good connection between Graph track position and Points position. If they could be connected, that would be a massive improvement in usability for track editing. i.e. select a point via graph view and the Points list scrolls to that point, and in reverse, if you select a point on the Points list it could position the Graph spot in that same position.

This could also be a good solution for scrolling the points list !!

**Comment by AndyB69** [comment]
Hello, why was this closed? It still needs resolution - Item 1 may be resolved, but item 2 is still an open issue, and item 3 can be resolved by implementing the suggestion [above](https://github.com/osmandapp/OsmAnd/issues/13749#issuecomment-1046791632).
